src/Microsoft.AspNetCore.Mvc.RazorPages/ApplicationModels/PageApplicationModel.cs

```diff
@@ -79,8 +91,33 @@ public PageApplicationModel(PageApplicationModel other)
         public IDictionary<object, object> Properties { get; }
 
         /// <summary>
-        /// Gets the <see cref="SelectorModel"/> instances.
+        /// Gets or sets the <see cref="TypeInfo"/> of the Razor page.
+        /// </summary>
+        public TypeInfo PageType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="TypeInfo"/> of the Razor page model.
+        /// </summary>
+        public TypeInfo ModelType { get; set; }
+
+        /// <summary>
+        /// Gets the <see cref="TypeInfo"/> of the handler.
+        /// </summary>
+        public TypeInfo HandlerType { get; }
+
+        /// <summary>
+        /// Gets the sequence of attributes declared on <see cref="HandlerType"/>.
+        /// </summary>
+        public IReadOnlyList<object> HandlerTypeAttributes { get; }
+
+        /// <summary>
+        /// Gets the sequence of <see cref="PageHandlerModel"/> instances.
+        /// </summary>
+        public IList<PageHandlerModel> HandlerMethods { get; }
+
+        /// <summary>
+        /// Gets the sequence of <see cref="PagePropertyModel"/> instances on <see cref="PageHandlerModel"/>.
         /// </summary>
-        public IList<SelectorModel> Selectors { get; }
+        public IList<PagePropertyModel> HandlerProperties { get; }
```

Comment:
ControllerModel has `ControllerProperties` (https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/ControllerModel.cs#L96). `BoundProperties` is on the ControllerAD.